package.json

```diff
@@ -177,11 +177,11 @@
   "bundlesize": [
     {
       "path": "lib/**/*.js",
-      "maxSize": "46 kB"
+      "maxSize": "48 kB"
```

Comment:
I think there's an opportunity to refactor so that we don't have to increase the max bundle size. If you take a look at the rendered templates and look at the generated typescript code for setting query parameters, we have an if/else statement for each query parameter. For example, this is what it looks like for `getProduct`:

```javascript
...
if (optionParams["select"] !== undefined) {
          queryParams["select"] = optionParams["select"];
        } else if (configParams["select"] !== undefined) {
          queryParams["select"] = configParams["select"];
        }
        if (optionParams["inventoryIds"] !== undefined) {
          queryParams["inventoryIds"] = optionParams["inventoryIds"];
        } else if (configParams["inventoryIds"] !== undefined) {
          queryParams["inventoryIds"] = configParams["inventoryIds"];
        }
        if (optionParams["currency"] !== undefined) {
          queryParams["currency"] = optionParams["currency"];
        } else if (configParams["currency"] !== undefined) {
          queryParams["currency"] = configParams["currency"];
        }
        if (optionParams["expand"] !== undefined) {
          queryParams["expand"] = optionParams["expand"];
        } else if (configParams["expand"] !== undefined) {
          queryParams["expand"] = configParams["expand"];
        }
        if (optionParams["locale"] !== undefined) {
          queryParams["locale"] = optionParams["locale"];
        } else if (configParams["locale"] !== undefined) {
          queryParams["locale"] = configParams["locale"];
        }
        if (optionParams["allImages"] !== undefined) {
          queryParams["allImages"] = optionParams["allImages"];
        } else if (configParams["allImages"] !== undefined) {
          queryParams["allImages"] = configParams["allImages"];
        }
...
```

But now that we expose the `ParamKeys`, we could theoretically do something like:

```javascript
getProductParamKeys.forEach(key => {
        if (optionParams[key] !== undefined) {
          queryParams[key] = optionParams[key];
        } else if (configParams[key] !== undefined) {
          queryParams[key] = configParams[key];
        }
})
```
There's probably going to be additional refactoring to ensure the proper separation of query and path parameters, but this approach should help reduce the current bundle size

https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/blob/778305483021d5daeaaa5af71f332b8c5e3ec6b2/templates/operations.ts.hbs#L146

Reply:
@joeluong-sfcc good catch! Let me give that a try 👀

Reply:
I've tried and gone down the rabbit hole. But it turns out that the resulting _gzip_ size is very similar to before. It looks like gzip compression does a very good job already.

Reply:
So I'm thinking to revert back to 467bf49ff756efde2274d067222370fd3f123a83. It has much smaller change, compared to what I have now.